service-base/src/main/java/org/eclipse/hono/servicebase/HonoServiceBase.java

```diff
@@ -21,36 +21,47 @@
 
 import java.util.Objects;
 
-public abstract class HonoServerBase extends AbstractVerticle {
+public abstract class HonoServiceBase extends AbstractVerticle {
 
-    private static final Logger LOG = LoggerFactory.getLogger(HonoServerBase.class);
+    private static final Logger LOG = LoggerFactory.getLogger(HonoServiceBase.class);
 
     protected HonoConfigProperties config = new HonoConfigProperties();
 
     protected int port         = Constants.PORT_UNCONFIGURED;
     protected int insecurePort = Constants.PORT_UNCONFIGURED;
 
+    /**
+     * Define the default value for the TLS port number in the implementing server class (e.g. 5671 for AMQPS 1.0).
+     * @return The default value for the TLS port.
+     */
     protected abstract int getPortDefaultValue();
+    /**
+     * Define the default value for the unencrypted port number in the implementing server class (e.g. 5672 for AMQP 1.0).
+     * @return The default value for the unencrypted port.
+     */
     protected abstract int getInsecurePortDefaultValue();
 
-    public int getPort() {
+    /**
+     * Return the port number of the TLS port.
```

Comment:
Very good point - maybe we can find a better naming when looking at the current state:
- HonoConfigProperties contain the "configured" port number
- HonoServiceBase contains a copy, but may change it during port "determination" (e.g. filling the default value)
- HonoServer may open an abitrary port number (if configured) and knows the REAL port number only after startup

Summarizing this I come to the conclusion:
- I will eliminate the method in the HonoServer
- therefore the server needs to update the port as soon as it is running (for ephemeral port configs).
Then we only have ONE method left and it always returns the real port number of the Hono Server when it is running.
Would let the name as it is then.

WDYT?